packages/backend/src/peripherals/database/ReportRepository.ts

```diff
@@ -48,10 +56,15 @@ export class ReportRepository extends BaseRepository {
   async addOrUpdateMany(reports: ReportRecord[]) {
```

Comment:
Maybe we could add a simple check to this function that all reports have the same timestamp?